### PR TITLE
Add Pulp 3 component architecture draft

### PR DESCRIPTION
**Pull request by ehelms**
This does a couple things aiming towards collaboration and centralization of architecture documentation.

 1) Introduces a `drafts/` folder for collaborative development while working towards being able to publish
 2) Adds a sub-folder for architecture documentation
 3) Adds a component folder and transfer Pulp 3 information from https://github.com/theforeman/puppet-pulpcore/blob/master/README.md#installation-layout

### CODE REVIEW

**Comment by melcorr** [comment]
You ok with this approach @maximiliankolb ? 
After this PR, I will merge #555 and move it into `/drafts/SOE` ?

File: drafts/architecture/components/pulp.adoc
Comment (ekohl) [inline drafts/architecture/components/pulp.adoc:16]: This link is 404 now and by now the projects are aligned again.

File: drafts/architecture/components/pulp.adoc
Comment (ekohl) [inline drafts/architecture/components/pulp.adoc:22]: I've also noticed that if the assets are not collected, the application refuses to start up. So they must be generated, they should be served (RFC2119 usage of must/should).

**Comment by maximiliankolb** [comment]
> You ok with this approach @maximiliankolb ?
> After this PR, I will merge #555 and move it into `/drafts/SOE` ?

Yes, having a `drafts/` directory for this purpose sounds good!

**Comment by melcorr** [comment]
Whenever @ekohl expresses satisfaction, I am happy for this to be merged.

**Comment by melcorr** [comment]
@ekohl  any objections to merging this?

**Comment by melcorr** [comment]
This is over three weeks waiting on a review. 
If I don't get a response by the end of the week I will merge as it doesn't affect anything live and is holding up another PR I want to merge.

**Comment by ekohl** [comment]
To be fair, there was a week PTO in that. I've not had the time to read through it yet.

**Comment by melcorr** [comment]
Can we set some deadline for this? 
I would like to create this drafts section so I can merge the SEO PR and migrate that to drafts as well. 
Since this won't be published anywhere, it is of lesser risk for now to merge it. 
I didn't merge on Friday, but now we're moving into week 4, week 3 including PTO.

**Comment by ares** [comment]
That's been another week, I believe it can be merged. If it's a priority, it can be reviewed after the merge too and fixed later if necessary.

**Review by maximiliankolb — APPROVED** [review]
It goes to "drafts/", so I agree with @ares

File: drafts/architecture/components/pulp.adoc
Comment (ekohl) [inline drafts/architecture/components/pulp.adoc:5]: This looks copied from our module documentation. `this module` is ambiguous.

**Review by ekohl — CHANGES_REQUESTED** [review]
I finally had some time to read through this. What I don't like about including this now is that it's a markdown to asciidoc conversion of puppet-pulpcore's README. Since it's a copy, it can go out of date. Wouldn't it be better to link to https://github.com/theforeman/puppet-pulpcore/blob/master/README.md or even tell users to read `/usr/share/foreman-installer/modules/pulpcore/README.md`?

What would make sense if it talked more about how puppet-foreman_proxy_content lays things out. That exposes fewer parameters (you can't change the user, most paths are not configurable. Mentioning `$apache_docroot` is just confusing if no user will ever see this.

**Comment by lzap** [comment]
What is the status? I am not against a new drafts/ folder, however, we could actually create a guide named Drafts or whatever and then add a Draft ribbon to the top left corner. This would be much more visible to users so they could provide feedback:

https://github.com/theforeman/foreman-documentation/pull/710/files

**Comment by calvingsmith** [comment]
Has there been any decision on this? Reading it I'm not sure where we are at at this point.

**Comment by maximiliankolb** [comment]
> Has there been any decision on this? Reading it I'm not sure where we are at at this point.

IMHO we kinda agreed on a `drafts/` folder, which is exactly what this PR does. So +1 from me. cc @lzap @melcorr

**Comment by melcorr** [comment]
I've asked a few times for this to be merged.

**Pull request merged**
